Factory/CatalogPermissionsFactory.php

```diff
@@ -53,6 +46,11 @@ public function getCatalogPermissionsHelper()
         return $this->objectManager->create('\Magento\CatalogPermissions\Helper\Data');
     }
 
+    public function getCatalogPermissionsConfig()
+    {
+        return $this->objectManager->create('\Magento\CatalogPermissions\App\Config');
```

Comment:
Is there any particular reason why the objectManager is used in these 2 factories rather than construct injection ?

Reply:
Perhaps I'm not using constructors enough to know but I used the objectManager to generate the classes because catalogpermissions is a Commerce feature and isn't available on open source. I don't want to generate errors if the module doesn't exists or isn't enabled. I'm free to discuss if you have an improved idea for this case.